[Streams][Significant events] Add knowledge indicators docs (#6223)
elastic/docs-content: solutions/observability/streams/management/knowledge-indicators.md

applies_to:
  serverless: preview
  stack: preview 9.4+
products:
  - id: observability
  - id: elasticsearch
  - id: kibana
  - id: cloud-serverless
  - id: cloud-hosted
  - id: cloud-enterprise
  - id: cloud-kubernetes
  - id: elastic-stack
---

# Knowledge Indicators [streams-knowledge-indicators]

Knowledge Indicators (KIs) are structured facts that Elastic extracts from your raw log data automatically without requiring schemas, service catalogs, or manual configuration. When you run extraction against a log stream, Elastic analyzes the raw data and returns facts about your environment: which services are running, the underlying infrastructure they rely on, how they depend on each other, and the log schemas they use.

Rather than a static configuration, this knowledge accumulates over time, automatically expires when a service disappears, and feeds directly into downstream capabilities like Rules, topology maps, AI agent investigations, and dashboards.

To access Knowledge Indicators, open **Significant Events** from the Streams main page and select the **Knowledge Indicators** tab.

:::{admonition} Requirements
To use this feature, you need:

> - A [Generative AI connector](kibana://reference/connectors-kibana/gen-ai-connectors.md). Using one can incur additional costs.

Author: ```suggestion
- A [Generative AI connector](kibana://reference/connectors-kibana/gen-ai-connectors.md), which can incur additional costs.
```

> - The `observability:streamsEnableSignificantEvents` {{kib}} settings enabled.

Author: ```suggestion
- The `observability:streamsEnableSignificantEvents` {{kib}} setting enabled.
```

  - id: observability
  - id: elasticsearch
  - id: kibana
  - id: cloud-serverless
  - id: cloud-hosted
  - id: cloud-enterprise
  - id: cloud-kubernetes
  - id: elastic-stack
---

# Knowledge Indicators [streams-knowledge-indicators]

:::{note}
This feature requires a [Generative AI connector](kibana://reference/connectors-kibana/gen-ai-connectors.md) and can incur additional costs.
:::

Knowledge Indicators (KIs) are structured facts that Elastic extracts from your raw log data automatically without requiring schemas, service catalogs, or manual configuration. When you run extraction against a log stream, Elastic analyzes the raw data and returns facts about your environment: which services are running, the underlying infrastructure they rely on, how they depend on each other, and the log schemas they use.

Rather than a static configuration, this knowledge accumulates over time, automatically expires when a service disappears, and feeds directly into downstream capabilities like Rules, topology maps, AI agent investigations, and dashboards.

To access Knowledge Indicators, open **Significant Events** from the Streams main page and select the **Knowledge Indicators** tab.

> :::{admonition} Required settings

Reviewer: Consider the following alternative:
Combine the note about the Generative AI connector with the "Required settings" into a single section "Requirements" and put everything after the introductory paragraphs.

To access the Significant Events page and generate KIs, you need to enable the following Kibana settings:
- `observability:streamsEnableSignificantEvents`
- `observability:streamsEnableSignificantEventsDiscovery`
:::

## Generate Knowledge Indicators [streams-ki-generate]

You can trigger KI extraction on demand or set up continuous extraction at a specific interval.

On demand
:   From the **Significant Events** page, select the streams you want to generate KIs for and click **Generate**.

Continuous extraction
:   When enabled, continuous extraction runs automatically on managed streams at the interval you configure. Continuous extraction is off by default. To enable it:

    1. From the **Streams** main page, navigate to **Significant Events** -> **Settings**.
   1. Under **Continuous KI extraction**, turn on **Enable continuous KI extraction**.
   1. Set the **Extraction interval** in hours, and list any **Excluded streams** to skip during continuous extraction.

> ## How extraction works [streams-ki-extraction]

Reviewer: ```suggestion
## How KI extraction works [streams-ki-extraction]
```
A more specific title might help SEO

Extraction runs entirely as a background task and never blocks ingestion. You can trigger it on demand from the stream detail view or the Significant Events Discovery UI.

## KI types [streams-ki-types]

Knowledge Indicators fall into two categories: Feature KIs and Query KIs.

### Feature KIs [streams-ki-feature]

Feature KIs are descriptive and explain the contents of the stream: what services are running, the infrastructure housing them, their dependencies, and the active tech stack.

Feature KIs carry a full data model:

- **`type` / `subtype`**: the category of the fact (Entity, Infrastructure, Technology, Dependency, Schema)
- **`title` / `description`**: a human-readable summary
- **`properties`**: stable key-value pairs used to deduplicate findings across multiple runs
- **`confidence`**: 0–100. LLM-identified KIs score based on evidence quality. Deterministic KIs always score 100.
- **`evidence`**: 2–5 supporting log excerpts that justify the KI's existence

> - **`filter`**: an optional [StreamLang](./streamlang.md) condition scoping the KI to specific documents

Reviewer: Minor edit: Start with a capital letter after a colon.

  "severity_score": 90,
  "esql": {
    "query": "FROM logs-* | WHERE service.name == \"postgres\" AND message : \"remaining connection slots\""
  }
}
```

## Downstream uses [streams-ki-uses]

KIs serve as the contextual foundation for several capabilities:

**Rules**: Query KIs automatically generate active Rules to surface signals without manual configuration.

**Topology graphs**: Dependency KIs construct an infrastructure graph inferred entirely from log data — no distributed tracing or manual configuration required. During an incident, the graph immediately shows which upstream services are affected when a specific database or service goes down.

**AI agent investigations**: Instead of reconstructing basic facts during every incident, an AI agent begins with your system's actual topology and known failure modes. It identifies the relevant streams, runs the applicable queries, and formulates a specific hypothesis using the available KI context.

**Dashboards**: KIs drive AI-generated dashboard suggestions for your streams.

**Grok patterns**: KIs inform Grok pattern generation when you introduce new streams.

> ## Lifecycle and maintenance [streams-ki-lifecycle]

Reviewer: ```suggestion
## KI Lifecycle and maintenance [streams-ki-lifecycle]
```
